macrosynergy/panel/view_timelines.py

```diff
@@ -147,195 +127,100 @@ def view_timelines(
             .cumsum()
         )
 
-    max_plots: int = 1
-    if (len(cids) == 1) and xcat_grid:
-        max_plots = len(xcats)
-    elif (len(cids) > 1) and not single_chart:
-        max_plots = len(cids)
-    ncol: int = min(ncol, max_plots)
-
-    sns.set(style="darkgrid")
-    cross_mean: Optional[pd.DataFrame] = None
-    cs_label: str = f"cross-sectional average of {xcats[0]}."
-    if cs_mean and (len(cids) > 1):
-        dfw: pd.DataFrame = df.pivot(index="real_date", columns="cid", values="value")
-        cm_series: pd.Series = dfw.mean(axis=1)
-        cross_mean: pd.DataFrame = pd.DataFrame(
-            data=cm_series.to_numpy(), index=cm_series.index, columns=["average"]
-        ).reset_index(level=0)
+    
+    face_plotter: FacetPlot = FacetPlot(
+        df=df, cids=cids, xcats=xcats, metrics=[val], start_date=start, end_date=end
+    )
+    line_plotter: LinePlot = LinePlot(
+        df=df, cids=cids, xcats=xcats, metrics=[val], start_date=start, end_date=end
+    )
```

Comment:
Why open an instance of both `FacetPlot` and `LinePlot` when we only need one (depending on how many cross sections we have)?

Wouldn't a better notation be to have:
```python
if xcat_grid or not single_chart:
    plotter: FacetPlot = FacetPlot(
        df=df, cids=cids, xcats=xcats, metrics=[val], start_date=start, end_date=end
    )
else:
    plotter: LinePlot = LinePlot(
        df=df, cids=cids, xcats=xcats, metrics=[val], start_date=start, end_date=end
    )
```

Reply:
Assuming the both have a shared base class of `Plotter` or similar.

Reply:
Kindly ignore, old code